observability/src/main/java/io/grpc/observability/logging/CloudLoggingHandler.java

```diff
@@ -0,0 +1,190 @@
+import java.util.logging.ErrorManager;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+
+/**
+ * Custom logging handler that outputs logs generated using {@link java.util.logging.Logger} to
+ * Cloud Logging.
+ */
+@Internal
+public class CloudLoggingHandler extends Handler {
+
+  private static final String DEFAULT_LOG_NAME = "grpc-observability";
+  private static final Level DEFAULT_LOG_LEVEL = Level.ALL;
+
+  private final LoggingOptions loggingOptions;
+  private final Logging loggingClient;
+  private final Level baseLevel;
+  private final String cloudLogName;
+
+  /**
+   * Creates a custom logging handler that publishes message to Cloud logging. Default log level is
+   * set to Level.FINEST if level is not passed.
+   */
+  public CloudLoggingHandler() {
+    this(DEFAULT_LOG_LEVEL, null, null);
+  }
+
+  /**
+   * Creates a custom logging handler that publishes message to Cloud logging.
+   *
+   * @param level set the level for which message levels will be logged by the custom logger
+   */
+  public CloudLoggingHandler(Level level) {
+    this(level, null, null);
+  }
+
+  /**
+   * Creates a custom logging handler that publishes message to Cloud logging.
+   *
+   * @param level set the level for which message levels will be logged by the custom logger
+   * @param logName the name of the log to which log entries are written
+   */
+  public CloudLoggingHandler(Level level, String logName) {
+    this(level, logName, null);
+  }
+
+  /**
+   * Creates a custom logging handler that publishes message to Cloud logging.
+   *
+   * @param level set the level for which message levels will be logged by the custom logger
+   * @param logName the name of the log to which log entries are written
+   * @param overrideProjectId the value of cloud project id to which logs are sent to by the custom
+   *     logger
+   */
+  public CloudLoggingHandler(Level level, String logName, String overrideProjectId) {
+    baseLevel = level.equals(DEFAULT_LOG_LEVEL) ? Level.FINEST : level;
+    setLevel(baseLevel);
+
+    cloudLogName = logName != null ? logName : DEFAULT_LOG_NAME;
+
+    try {
+      // TODO(dnvindhya) read the value from config instead of taking it as an argument
+      String projectId = overrideProjectId != null ? overrideProjectId : null;
+
+      if (!Strings.isNullOrEmpty(projectId)) {
+        loggingOptions = LoggingOptions.newBuilder().setProjectId(overrideProjectId).build();
+      } else {
+        loggingOptions = LoggingOptions.getDefaultInstance();
+      }
+
+      loggingClient = loggingOptions.getService();
+    } catch (Exception ex) {
+      reportError("Error in initializing Cloud Logging", ex, ErrorManager.OPEN_FAILURE);
+      throw ex;
+    }
+  }
+
+  @Override
+  public void publish(LogRecord record) {
+    if (!(record instanceof LogRecordExtension)) {
+      throw new IllegalArgumentException("Expected record of type LogRecordExtension");
+    }
+
```

Comment:
Unless it is serving some purpose (like delineating major pieces of code) these blank lines can be removed